src/main/scala/is/hail/variant/HTSGenotypeView.scala

```diff
@@ -49,7 +50,7 @@ sealed abstract class HTSGenotypeView {
 
 class TGenotypeView(rs: TStruct) extends HTSGenotypeView {
   private val tgs = rs.fieldType(3).asInstanceOf[TArray]
-  private val tg = TGenotype.representation
+  private val tg = TGenotype().representation
```

Comment:
This should reference the object, not a throwaway allocation of the case class

Reply:
Actually I'm probably wrong about this. We should specifically be using a non-required genotype.

Reply:
1. This is really confusing since I thought I changed it back---maybe i did somewhere else?
2. TGenotype() and friends don't actually allocate--the apply method returns the singletons

Reply:
ah I'm wrong about the allocation then, regardless this should stay as it is.

Reply:
I also moved TGenotype.representation back from the companion object in to the class, since it doesn't make sense to define it on the companion object which doesn't have a notion of requiredness when instead it could be defined on the two case objects (which do)